internal/ingress/annotations/class/main_test.go

```diff
@@ -62,7 +65,142 @@ func TestIsValidClass(t *testing.T) {
 		IngressClass = test.controller
 		DefaultClass = test.defClass
 
-		b := IsValid(ing)
+		b := IsValid(&ing.ObjectMeta)
+		if b != test.isValid {
+			t.Errorf("test %v - expected %v but %v was returned", test, test.isValid, b)
+		}
+	}
+}
+
+func TestIsValidPlugin(t *testing.T) {
```

Comment:
Are these tests necessary to be repeated for each type since the method now takes in ObjectMeta?
If not, let remove those as additional tests are redundant.